service/controller/resource/heartbeatrouting/create.go

```diff
@@ -0,0 +1,49 @@
+package heartbeatrouting
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/prometheus-meta-operator/service/controller/resource/heartbeatrouting/receiver"
+	"github.com/giantswarm/prometheus-meta-operator/service/controller/resource/heartbeatrouting/route"
+	"github.com/giantswarm/prometheus-meta-operator/service/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	r.logger.LogCtx(ctx, "level", "debug", "message", "checking if alertmanager configmap needs to be updated")
+
+	cluster, err := key.ToCluster(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	configMap, cfg, err := r.readConfig(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	cfg, receiverUpdated, err := receiver.EnsureCreated(cfg, cluster, r.installation, r.opsgenieKey)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	cfg, routeUpdated, err := route.EnsureCreated(cfg, cluster, r.installation)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if receiverUpdated || routeUpdated {
```

Comment:
Do we need a negation here?

e.g: `if !receiverUpdated || !routeUpdate`? Otherwise it sounds like we update it if we updated it

Reply:
Or having the booleans named the other way around (i.e: receiverNeedsUpate)

Reply:
I think this could be renamed for sure. Basically, those imply that the alertmanagerconfig struct changed and the configmap needs to be updated. receiverUpdated would mean that one receiver was changed or was added to the list of receiver so we need to update the configmap to reflect the change in alertmanager

Reply:
This is fixed in the latest commit